clang-tools-extra/clang-doc/JSONGenerator.cpp

```diff
@@ -277,9 +282,30 @@ static void serializeInfo(const TemplateInfo &Template, Object &Obj) {
     TemplateObj["Parameters"] = ParamsArray;
   }
 
+  if (!Template.Constraints.empty()) {
+    json::Value ConstraintsArray = Array();
+    auto &ConstraintsArrayRef = *ConstraintsArray.getAsArray();
+    ConstraintsArrayRef.reserve(Template.Constraints.size());
+    for (const auto &Constraint : Template.Constraints) {
+      json::Value ConstraintVal = Object();
+      auto &ConstraintObj = *ConstraintVal.getAsObject();
+      serializeInfo(Constraint, ConstraintObj);
+      ConstraintsArrayRef.push_back(ConstraintVal);
+    }
```

Comment:
We use this code pattern so much, I'm wondering if we should have an API for it ... maybe even one in `json`...

Reply:
Refactored for this patch as a function, I can refactor other parts later.